ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -208,12 +205,11 @@ def start_learning(self, env_manager: EnvManager) -> None:
         global_step = 0
         last_brain_behavior_ids: Set[str] = set()
         try:
-            initial_step = self._reset_env(env_manager)
             # Create the initial set of trainers and managers
             initial_brain_behaviors = set(env_manager.external_brains.keys())
             self._create_trainers_and_managers(env_manager, initial_brain_behaviors)
             last_brain_behavior_ids = initial_brain_behaviors
-            self._process_step_infos(initial_step)
+            self._reset_env(env_manager)
```

Comment:
It seems slightly odd that whole env gets reset when a new agent name is encountered.  It seems like this would end the trajectories of the other agents too, even if they are not in a done state. Is that correct?

Reply:
This only happens at the first step - the rest of the steps are in the while True loop